presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -467,4 +473,31 @@ public void testPickTableLayoutWithFilter()
                         filter("orderkey = BIGINT '5'",
                                 constrainedTableScanWithTableLayout("orders", filterConstraint, ImmutableMap.of("orderkey", "orderkey")))));
     }
+
+    @Test
+    public void testBroadcastCorrelatedSubqueryAvoidsRemoteExchangeBeforeAggregation()
+    {
+        Session broadcastJoin = Session.builder(this.getQueryRunner().getDefaultSession())
+                .setSystemProperty(DISTRIBUTED_JOIN, Boolean.toString(false))
+                .setSystemProperty(FORCE_SINGLE_NODE_OUTPUT, Boolean.toString(false))
+                .build();
+        // region is unpartitioned, AssignUniqueId should provide satisfying partitioning for count(*) after LEFT JOIN
+        assertEquals(
+                countOfMatchingNodes(
+                        plan("SELECT (SELECT count(*) FROM region r2 WHERE r2.regionkey > r1.regionkey) FROM region r1", OPTIMIZED_AND_VALIDATED, broadcastJoin, false),
+                        node -> node instanceof ExchangeNode && ((ExchangeNode) node).getScope() == REMOTE),
+                // one RemoteExchange for LEFT JOIN build rows
```

Comment:
Checking for total number of RemoteExchange nodes doesn't feel enough. Would it be possible to check that the exchange node is indeed "for LEFT JOIN build rows"?

Reply:
The plan is quite complex and join must have remote exchange for at least one source until it colocated (which it isn't)